3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/PiiPseudonymizerEnrichment.scala

```diff
@@ -0,0 +1,237 @@
+      case JObject(List(("json", jsonField))) =>
+        val fieldsDisj: \/[MappingException, List[String]] = List(extractString(jsonField, "field"),
+                                                                  extractString(jsonField, "schemaCriterion"),
+                                                                  extractString(jsonField, "jsonPath")).sequenceU
+        fieldsDisj.validation
+          .leftMap(_ => "Configuration file has unexpected structure")
+          .flatMap[String, PiiJson] { (l: List[String]) =>
+            SchemaCriterion
+              .parse(l(1))
+              .rightMap { sc =>
+                PiiJson(strategy, l(0), sc, l(2))
+              }
+              .leftMap(_.getMessage)
+          }
+      case x => "Configuration file has unexpected structure".failure
+    }.sequenceU
+
+  implicit val json4sFormats = DefaultFormats
+
+  private def extractString(jValue: JValue, field: String): \/[MappingException, String] =
+    \/.fromTryCatchThrowable[String, MappingException] {
+      (jValue \ field).extract[String]
+    }
+
+  private def extractStrategyFunction(config: JValue) =
+    ScalazJson4sUtils
+      .extract[String](config, "parameters", "strategy", "pseudonymize", "hashFunction")
+      .leftMap(_.getMessage)
+
+  private def matchesSchema(config: JValue, schemaKey: SchemaKey): Validation[String, JValue] =
+    if (supportedSchema.matches(schemaKey)) {
+      config.success
+    } else {
+      ("Schema key %s is not supported. A '%s' enrichment must have schema '%s'.")
+        .format(schemaKey, supportedSchema.name, supportedSchema)
+        .failure
+    }
+}
+
+/**
+ * The PiiPseudonymizerEnrichment runs after all other enrichments to find fields that are configured as PII and apply
+ * some anonymization (currently only psudonymizantion) on them. Currently a single strategy for all the fields is
+ * supported due to the config format, and there is only one implemented strategy, however the enrichment supports a
+ * strategy per field configuration.
+ *
+ * The user may specify two types of fields POJO or JSON. A POJO field is effectively a scalar field in the
+ * EnrichedEvent, whereas a JSON is a "context" formatted field (a JSON string in "contexts" field in enriched event)
+ *
+ * @param fieldList a lits of configured PiiFields
+ */
+case class PiiPseudonymizerEnrichment(fieldList: List[PiiField]) extends Enrichment {
+  def transformer(transformMap: TransformMap): TransformMap = transformMap ++ fieldList.map(_.transformer(transformMap)).reduce(_ ++ _)
+}
+
+/**
+ * Specifies a field in POJO and the strategy that should be applied to it.
+ * @param strategy the strategy that should be applied
+ * @param fieldName the field where the strategy will be applied
+ */
+final case class PiiPojo(strategy: PiiStrategy, fieldName: String) extends PiiField {
+  override def applyStrategy(fieldValue: String): String = strategy.scramble(fieldValue)
```

Comment:
I'm not sure I understand your usage pattern for `override` modifier. People either use it only when overriding some default behavior (like me) or when overriding anything defined in parent `trait`/`class` like @BenFradet, but here I see it randomly.

Reply:
I don't understand exactly what you mean. It implements a trait. Random overrides don't compile.

Reply:
`supportedSchema` in `PiiPseudonymizerEnrichment` also implements a trait, but doesn't have a modifier. I just thought that `parse` is method of `ParseableEnrichment` as well, but turns out it is missing only in `supportedSchema`.